src/Compilers/CSharp/Portable/Symbols/Source/SourceModuleSymbol.cs

```diff
@@ -619,6 +620,55 @@ internal override void AddSynthesizedAttributes(ModuleCompilationState compilati
             }
         }
 
+        // PROTOTYPE(NullableReferenceTypes): Handle NullableAttribute
+        // consistently with other embedded attributes.
+        internal NamedTypeSymbol GetNullableAttribute(DiagnosticBag diagnostics = null)
+        {
+            if (UtilizesNullableReferenceTypes && (object)_lazyNullableAttribute == null)
+            {
+                Debug.Assert(diagnostics != null); // First caller should be prepared to handle diagnostics.
+
+                var bag = DiagnosticBag.GetInstance();
+                var symbol = new SynthesizedEmbeddedAttributeSymbol(
+                    _assemblySymbol.DeclaringCompilation,
+                    AttributeDescription.NullableAttribute.Namespace,
+                    AttributeDescription.NullableAttribute.Name,
+                    GetNullableAttributeMembers,
+                    bag);
+                if ((object)Interlocked.CompareExchange(ref _lazyNullableAttribute, symbol, null) == null)
+                {
+                    diagnostics.AddRange(bag);
+                }
+                bag.Free();
+            }
+
+            return _lazyNullableAttribute;
+        }
+
+        private static ImmutableArray<Symbol> GetNullableAttributeMembers(
+            CSharpCompilation compilation,
+            NamedTypeSymbol containingType,
+            DiagnosticBag diagnostics)
+        {
+            var boolType = compilation.GetSpecialType(SpecialType.System_Boolean);
+            var info = boolType.GetUseSiteDiagnostic();
```

Comment:
>var info = boolType.GetUseSiteDiagnostic(); [](start = 12, length = 43)

Consider using Binder.ReportUseSiteDiagnostics helper here, similar to the code in SynthesizedEmbeddedAttributeSymbol constructor.

Reply:
Done.